pytest_html/resources/main.js

```diff
@@ -190,3 +190,30 @@ function one_row_for_data() {
         }
     });
 }
+
+function filter_table(elem) {
+    console.log(elem.value);
+    var checkbox = elem;
+    var att_name = "data-type-associated";
+
+    var results_table = document.getElementById("results-table");
+    var rows = results_table.rows; 
+
+    //The logs and extra HTML don't appear on the rows
+    var logs = document.getElementsByClassName("extra");
+
+    
+    for(var i = 0; i < rows.length; i++){
+        if(rows[i].getAttribute(att_name) ==
+            elem.getAttribute(att_name)) {
+            rows[i].hidden = !checkbox.checked;
+        }
+    }
+
+    for(var i = 0; i < logs.length; i++) {
```

Comment:
The inner code is identical though, so if you combined `rows` and `logs` and then iterated through them, wouldn't that work?